src/driveformatthread.cpp

```diff
@@ -63,8 +67,67 @@ void DriveFormatThread::run()
     }
 #endif
 
+#ifdef Q_OS_WIN
+    // Windows-specific disk preparation
+    emit preparationStatusUpdate(tr("Preparing disk for formatting..."));
+    
+    std::regex windriveregex("\\\\\\\\.\\\\PHYSICALDRIVE([0-9]+)", std::regex_constants::icase);
+    std::cmatch m;
+
+    if (std::regex_match(_device.constData(), m, windriveregex))
+    {
+        QByteArray diskNumber = QByteArray::fromStdString(m[1]);
+        
+        // Check for mounted volumes
+        auto l = Drivelist::ListStorageDevices();
+        QByteArray devlower = _device.toLower();
+        bool hasVolumes = false;
+        
+        for (auto i : l)
+        {
+            if (QByteArray::fromStdString(i.device).toLower() == devlower)
+            {
+                hasVolumes = !i.mountpoints.empty();
+                break;
+            }
+        }
+
+        // Clean disk with diskpart (required for mounted volumes, safe for clean disks)
+        emit preparationStatusUpdate(tr("Cleaning disk..."));
+        QProcess diskpartProcess;
+        diskpartProcess.start("diskpart", QStringList());
+        if (!diskpartProcess.waitForStarted(5000))
+        {
+            emit error(tr("Failed to start disk cleanup utility. Please ensure you have administrator privileges."));
+            return;
+        }
+        
+        QString script = QString("select disk %1\nclean\nrescan\n").arg(diskNumber);
+        diskpartProcess.write(script.toLatin1());
+        diskpartProcess.closeWriteChannel();
+        
+        if (!diskpartProcess.waitForFinished(60000))
```

Comment:
Different timeout value here (60 seconds) to in `downloadthread.cpp` (30 seconds)?
This is probably a stupid question, but why does `downloadthread.cpp` retry diskpart up to 3 times, but `driveformathread.cpp` only does it once?
Even stupider question: I wonder if it might be worth doing a small refactoring so that the diskpart-related code isn't "duplicated" in different places?